webui/src/main/java/org/polypheny/db/webui/crud/LanguageCrud.java

```diff
@@ -83,7 +84,64 @@ public LanguageCrud( Crud crud ) {
     public Result processCqlRequest( Session session, QueryRequest request ) {
         try {
             String cqlQueryStr = request.query;
-            if ( cqlQueryStr.equals( "" ) ) {
+            if ( cqlQueryStr.trim().equals( "" ) ) {
+                throw new RuntimeException( "CQL query is an empty string!" );
+            }
+
+            CqlParser cqlParser = new CqlParser( cqlQueryStr, "APP" );
+            CqlQuery cqlQuery = cqlParser.parse();
+
+            if ( log.isDebugEnabled() ) {
+                log.debug( "Starting to process CQL resource request. Session ID: {}.", session );
+            }
+            //requestCounter.incrementAndGet();
+            Transaction transaction = this.crud.getTransaction( request.analyze, request.cache );
+
+            if ( request.analyze ) {
+                transaction.getQueryAnalyzer().setSession( session );
+            }
+
+            // This is not a nice solution. In case of a sql script with auto commit only the first statement is analyzed
+            // and in case of auto commit of, the information is overwritten
+            InformationManager queryAnalyzer = null;
+            if ( request.analyze ) {
+                queryAnalyzer = transaction.getQueryAnalyzer().observe( crud );
+            }
+
+            Statement statement = transaction.createStatement();
+            RelBuilder relBuilder = RelBuilder.create( statement );
+            JavaTypeFactory typeFactory = transaction.getTypeFactory();
+            RexBuilder rexBuilder = new RexBuilder( typeFactory );
+
+            long executionTime = System.nanoTime();
+
+            Cql2RelConverter cql2RelConverter = new Cql2RelConverter( cqlQuery );
+
+            RelRoot relRoot = cql2RelConverter.convert2Rel( relBuilder, rexBuilder );
+            PolyphenyDbSignature<?> signature = statement.getQueryProcessor().prepareQuery( relRoot );
+
+            Result result = getResult( SchemaType.RELATIONAL, LanguageType.CQL, statement, request, cqlQueryStr, signature, request.noLimit );
+            try {
+                statement.getTransaction().commit();
+            } catch ( TransactionException e ) {
+                throw new RuntimeException( "Error while committing.", e );
+            }
+            executionTime = System.nanoTime() - executionTime;
+            if ( queryAnalyzer != null ) {
+                Crud.attachQueryAnalyzer( queryAnalyzer, executionTime );
+            }
+
+            return result;
+        } catch ( Exception e ) {
+            throw new RuntimeException( e );
+        }
+    }
+
+
+    public Result processCqlRequest( Session session, QueryRequest request, Crud crud ) {
```

Comment:
@datomo This method seems to be redundant

Reply:
yes, I have removed it